src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1339,13 +1342,13 @@ public virtual SortKey GetSortKey(string source)
             return CreateSortKey(source, CompareOptions.None);
         }
 
-        public override bool Equals(object value)
+        public override bool Equals(object? value)
         {
             return value is CompareInfo otherCompareInfo
                 && Name == otherCompareInfo.Name;
         }
 
-        public override int GetHashCode() => Name.GetHashCode();
+        public override int GetHashCode() => Name?.GetHashCode() ?? 0;
```

Comment:
Is this `?? 0` the general approach you guys are choosing for chained hash code like this?

Reply:
I think this is the first occurrence - I thought it is better than NRE :smile:

Reply:
also already wrote a note for that in here: https://github.com/dotnet/corefx/issues/36384

Reply:
name is no longer nullable right?

Reply:
Why is this changing?

> I thought it is better than NRE

In what situation today does it null ref?

Reply:
it actually should not be nullable per other comment so will be removing that